charts/steward/templates/config-pipelineruns.yaml

```diff
@@ -96,7 +96,18 @@ data:
   limitRange: {{ default ( .Files.Get "data/pipelineruns-default-limitrange.yaml" ) .Values.pipelineRuns.limitRange | quote }}
   resourceQuota: {{ .Values.pipelineRuns.resourceQuota | quote }}
 
-{{- with .Values.pipelineRuns.jenkinsfileRunner.podSecurityContext }}
+{{- with .Values.pipelineRuns.jenkinsfileRunner }}
+{{- if kindIs "string" .image }}
+  jenkinsfileRunner.image: {{ .image | quote }}
+  jenkinsfileRunner.imagePullPolicy: {{ .imagePullPolicy | quote }}
+{{- else }} 
+{{/* This is for backwards compatibility only */}}
+{{/* TODO: Remove in one of the next releases and add corresponding incompatibility notice to the changelog */}
+  jenkinsfileRunner.image: {{ printf "%s:%s" .image.repository .image.tag | quote }}
+  jenkinsfileRunner.imagePullPolicy: {{ .image.pullPolicy | quote }}
+{{- end -}}
```

Comment:
The `if` condition is always true because `jenkinsfileRunner.image` is set by default.

If a user has `jeninsfileRunner.image.repository` overridden, this value should have precedence over the default image.

Reply:
My statement was wrong. But we have a type conflict:

* `pipelineRuns.jenkinsfileRunner.image` is of type string now
* setting `pipelineRuns.jenkinsfileRunner.image.repository` would implicitly change `pipelineRuns.jenkinsfileRunner.image` to type object. Helm seems to support that but gives a warning.

Changing the type of `pipelineRuns.jenkinsfileRunner.image` seems to be dirty and may create further problems. We may not provide backward compatibility and fail in case the type is not string.

Reply:
Fail with an error if old values syntax is used. See 
2f0c56d